mbed-client-libservice/ns_trace.h

```diff
@@ -46,6 +46,74 @@
 extern "C" {
 #endif
 
+#define NS_TRACE_USE_MBED_TRACE
+#if defined(NS_TRACE_USE_MBED_TRACE)
+
+#if defined(HAVE_DEBUG)
+#ifndef FEA_TRACE_SUPPORT
+#define FEA_TRACE_SUPPORT
+#endif
+#endif
+
+#include "mbed-trace/mbed_trace.h"
+
+
+/* Convert libTrace calls to mbed-trace calls */
+#define trace_init(void)                    mbed_trace_init()
+#define trace_free(void)                    mbed_trace_free()
+#define set_trace_config(config)            mbed_trace_config_set(config)
+#define get_trace_config(void)              mbed_trace_config_get()
+#define set_trace_prefix_function(pref_f)   mbed_trace_prefix_function_set(pref_f)
+#define set_trace_suffix_function(suffix_f) mbed_trace_suffix_function_set(suffix_f)
+#define set_trace_print_function(print_f)   mbed_trace_print_function_set(print_f)
+#define set_trace_cmdprint_function(printf) mbed_trace_cmdprint_function_set(printf)
+#define set_trace_exclude_filters(filters)  mbed_trace_exclude_filters_set(filters)
+#define set_trace_include_filters(filters)  mbed_trace_include_filters_set(filters)
+#define get_trace_exclude_filters(void)     mbed_trace_exclude_filters_get()
+#define get_trace_include_filters(void)     mbed_trace_include_filters_get()
+#define trace_last(void)                    mbed_trace_last()
```

Comment:
Why `(void)` I do not see the need for it... These are not function definitions...

These macros now take 1 parameter with name `void`.
 Use `#define trace_last()  ...` to have a macro with no parameters.